Infrastructure/ExtensionsForInfrastructure.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using Core;
+
+namespace Infrastructure
+{
+    public static class ExtensionsForInfrastructure
+    {
+        public static IWebElement WaitAndFindElement(this ISearchContext context, By by)
```

Comment:
This is a bit too complicated, but a good idea! Look into WebDriverWait and ExpectedConditions, and maybe in the element properties themselves you'll be able to use the waiter